osbenchmark/aggregator.py

```diff
@@ -13,23 +13,31 @@ def __init__(self, cfg, test_executions_dict, args):
         self.args = args
         self.test_executions = test_executions_dict
         self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
-        self.accumulated_iterations: Dict[str, int] = {}
+        self.accumulated_iterations: Dict[str, Dict[str, int]] = {}
         self.metrics = ["throughput", "latency", "service_time", "client_processing_time", "processing_time", "error_rate", "duration"]
         self.test_store = metrics.test_execution_store(self.config)
         self.cwd = cfg.opts("node", "benchmark.cwd")
 
-    def count_iterations_for_each_op(self) -> None:
+    def count_iterations_for_each_op(self, test_execution) -> None:
         loaded_workload = workload.load_workload(self.config)
         test_procedure_name = self.config.opts("workload", "test_procedure.name")
         test_procedure_found = False
+        workload_params = {} if getattr(test_execution, 'workload_params') is None else test_execution.workload_params
+
+        test_execution_id = test_execution.test_execution_id
+        self.accumulated_iterations[test_execution_id] = {}
 
         for test_procedure in loaded_workload.test_procedures:
             if test_procedure.name == test_procedure_name:
                 test_procedure_found = True
```

Comment:
Super nit: Since `loaded_workload.test_procedures` is a list, we can reduce these three lines to:
```
if test_procedure_name in loaded_workload.test_procedures:
# Count iterations in each operation
else:
# raise value error
```

Reply:
hmm I don't think so because loaded_workload.test_procedures returns a list of test_procedure objects. So I'd have to match test_procedure_name to the name attribute in each of these objects

Reply:
Although I could use python's `next()` to search for the first object with a `name` attribute that matches the test_procedure_name. Something like:
```
matching_test_procedure = next((tp for tp in loaded_workload.test_procedures if tp.name == self.test_procedure_name), None)
if matching_test_procedure:
# count iterations
else:
# raise value error
```